src/aws-api-mcp-server/awslabs/aws_api_mcp_server/core/aws/service.py

```diff
@@ -0,0 +1,227 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import boto3
+import contextlib
+from ..aws.services import driver
+from ..common.config import AWS_API_MCP_PROFILE_NAME
+from ..common.errors import Failure
+from ..common.models import (
+    AwsApiMcpServerErrorResponse,
+    AwsCliAliasResponse,
+    Credentials,
+    InterpretationMetadata,
+    InterpretationResponse,
+    InterpretedProgram,
+    IRTranslation,
+    ProgramClassificationMetadata,
+    ProgramInterpretationResponse,
+    ProgramValidationResponse,
+)
+from ..common.models import Context as ContextAPIModel
+from ..common.models import ValidationFailure as FailureAPIModel
+from ..metadata.read_only_operations_list import (
+    ReadOnlyOperations,
+)
+from ..parser.lexer import split_cli_command
+from .driver import interpret_command as _interpret_command
+from botocore.exceptions import NoCredentialsError
+from io import StringIO
+from loguru import logger
+from typing import Any
+
+
+def get_local_credentials() -> Credentials:
+    """Get the local credentials for AWS profile."""
+    if AWS_API_MCP_PROFILE_NAME is not None:
+        session = boto3.Session(profile_name=AWS_API_MCP_PROFILE_NAME)
+    else:
+        session = boto3.Session()
+    aws_creds = session.get_credentials()
+
+    if aws_creds is None:
+        raise NoCredentialsError()
+
+    creds_source = (
+        aws_creds.method
+        if AWS_API_MCP_PROFILE_NAME is None
+        else f'{AWS_API_MCP_PROFILE_NAME} profile'
+    )
```

Comment:
Can you help explain why the `AWS_PROFILE` environmental variable is insufficient?
```suggestion
    session = boto3.Session()
    aws_creds = session.get_credentials()

    if aws_creds is None:
        raise NoCredentialsError()

    creds_source = aws_creds.method
```

Reply:
This was a suggestion from Clare, see this discussion https://github.com/awslabs/mcp/issues/481#issuecomment-2924127896